Unify Linux/FreeBSD build

anodos325 opened the pull request:
Bring in FreeBSD customizations. These will not compile on Linux (auto configured).

anodos325 committed:
s3:modules - Add FreeBSD-specific modules

Convert ixnas to use samba's generic NFSv4 ACL code
and update modules in general to use new VFS.

Convert vfs_zfs_space into a wrapper around zfs_core.

anodos325 committed:
lib/tevent - add tevent_kqueue for FreeBSD and update WAF

yocalebo reviewed (approved):
Once this is built, API tests should pick up any non-obvious failures with these changes....I hope 😉

rick-mesta commented on source3/modules/vfs_noacl.c:
```diff
@@ -0,0 +1,439 @@
+			return map_nt_error_from_unix(errno);
+		}
+		psbuf = &sbuf;
+        }
```
nit: wonky indentation

rick-mesta reviewed (approved):
several new files have 2020 copyright - expected ?

anodos325 committed:
s3:lib:system - add procfd paths for FreeBSD

FreeBSD 13 added O_PATH support. Similar behavior
to linux procfs fd paths may be obtained in
FreeBSD 13 and later by mounting the fdescfs
filesystem with the mount option 'nodup'. This
filesystem is not mounted by default. Expand the
list of potenial procfd paths to contain two
additional FreeBSD-related entries.
`/var/run/samba/fd` and `/dev/fd`. The former
is preferred because it allows the administrator
to maintain default freebsd fdescfs mount behavior
for /dev/fd (where opening the path is equivalent
to dup2()).

Signed-off-by: Andrew Walker <awalker@ixsystems.com>

anodos325 committed:
Update copyright in iX modules
